pkg/clusters/addons/kong/kong.go

```diff
@@ -146,11 +167,44 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		)
 	}
 
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
+	}
+
 	// do the deployment and install the chart
 	args := []string{"--kubeconfig", kubeconfig.Name(), "install", DefaultDeploymentName, "kong/kong"}
-	args = append(args, "--create-namespace", "--namespace", a.namespace)
-	args = append(args, a.deployArgs...)
+	args = append(args, "--namespace", a.namespace)
+	if a.enterprise {
+		if a.license != "" && a.license == KongEnterpriseTestLicense {
```

Comment:
This check highlights an issue with how we're handling the license secret name handling.

The logic here avoids deploying a license secret if the license secret name provided differs from the default. This suggests that we intend to allow users to provide existing secrets to license their kong proxy, however due to an interaction with the `clusters.CreateNamespace` function that was added as a part of this PR it will not be possible for an existing secret in the same namespace to be provided because the namespace creation method does not tolerate the namespace having already existed.

I recommend (but I'm open to your suggestions for alternative solutions) removing the `a.license` attribute entirely, and allowing this addon code to manage the lifecycle of the Kong enterprise license secret fully.

Reply:
In slack we discussed dropping the injectable license for now (can be added later) so this is all set :+1:

Reply:
This does not apply anymore since we are not handling license secret name anymore.